README.md

```diff
@@ -21,68 +27,147 @@ After installing and configuring those requirements, you can download [docker-co
+* `OIDC_GROUPS_CLAIM`: Name of claim to be used to represent groups (default: caluma_groups)
+
+#### Extension points
+
+Caluma is meant to be used as a service with a clean API hence it doesn't provide a Django app.
+For customization some clear extension points are defined. In case a customization is needed
+where no extension point is defined, best [open an issue](https://github.com/projectcaluma/caluma/issues/new) for discussion.
+
+##### Visibility classes
+
+Visibility classes define what user can read per node. Such can be be configured as `VISIBILITY_CLASSES`.
+
+Following pre-defined classes are available:
+* `caluma.core.visibilities.Any`: Allow any user without any filtering
+* `caluma.user.visibilities.Authenticated`: Filter results to none for not authenticated users
+* `caluma.user.visibilities.CreatedByGroup`: Filter to group node has been created for
+
+In case this default classes do not cover your use case, it is also possible to create your custom
+visibility class defining per node how to filter.
+
+Example:
+```python
+from caluma.types import Node
+from caluma.form.schema import Form
+
+class CustomVisibility(BaseVisibility):
+    @filter_queryset_for(Node)
+    def filter_queryset_for_all(self, node, queryset, info):
+        return queryset.filter(created_by_user=info.context.request.user.username)
+
+    @filter_queryset_for(Form)
+    def filter_queryset_for_form(self, node, queryset, info):
+        return queryset.exclude(slug='protected-form')
+"""
 
-```bash
-docker-compose build --pull
 ```
 
-### Setup pre commit
+Save your visibility module as `visibility.py` and inject it Docker container as volume to path `/app/caluma/extensions/visibility.py`.
 
-Pre commit hooks is an additional option instead of executing checks in your editor of choice.
+Afterwards you can configure it in `VISIBILITY_CLASSES` as `caluma.extensions.visibilities.CustomVisibility`.
 
-```bash
-# create virtualenv with tool of your choice
-pip install pre-commit
-pip install -r requiements-dev.txt -U
-pre-commit install
+##### Permission classes
+
+Permission classes define who may perform which mutation. Such can be configured as `PERMISSION_CLASSES`.
+
+Following pre-defined classes are available:
+* `caluma.user.permissions.IsAuthenticated`: only allow authenticated users
+* `caluma.core.permissions.AllowAny`: allow any users to perform any mutation.
+
+In case this default classes do not cover your use case, it is also possible to create your custom
+permission class defining per mutation and mutation object what is allowed.
+
+Example:
+```python
+from caluma.form.schema import SaveForm
+from caluma.mutation import Mutation
+
+class CustomPermission(BasePermission):
+    @permission_for(Mutation)
+    def has_permission_default(self, mutation, info):
```

Comment:
To meaningfully implement my own permission class/method, I need to know how to make my decision, ie. where to get my information. 
Please add some documentation explaining what's in the info parameter (I assume request, user, maybe more?) Also what does the mutation contain? Do I have access to the queryset?

Reply:
added some docs for arguments